EduNLP/Pretrain/quesNet_vec.py

```diff
@@ -0,0 +1,741 @@
+        save_vocab_path = os.path.join(tokenizer_config_dir, "vocab")
+        os.makedirs(save_vocab_path, exist_ok=True)
+        tokenizer_params = {
+            "img_token": self.img_token,
+            "unk_token": self.unk_token,
+            "pad_token": self.pad_token,
+            "max_length": self.max_length,
+            "meta": self.meta
+        }
+        self.save_vocab(save_vocab_path)
+        with open(tokenizer_config_path, "w", encoding="utf-8") as wf:
+            json.dump(tokenizer_params, wf, ensure_ascii=False, indent=2)
+
+    def padding(self, idx, max_length, type='word'):
+        padding_idx = idx + [self.stoi[type][self.pad_token]] * (max_length - len(idx))
+        return padding_idx
+
+    @property
+    def vocab_size(self):
+        return len(self.stoi['word'])
+
+    def set_img_dir(self, path):
+        self.img_dir = path
+
+
+def clip(v, low, high):
+    if v < low:
+        v = low
+    if v > high:
+        v = high
+    return v
+
+
+class lines:
+    def __init__(self, filename, skip=0, preserve_newline=False):
+        self.filename = filename
+        with open(filename):
+            pass
+        output = subprocess.check_output(('wc -l ' + filename).split())
+        self.length = int(output.split()[0]) - skip
+        self.skip = skip
+        self.preserve_newline = preserve_newline
+
+    def __len__(self):
+        return self.length
+
+    def __iter__(self):
+        for i in range(len(self)):
+            yield self[i]
+
+    def __getitem__(self, item):
+        d = self.skip + 1
+        if isinstance(item, int):
+            if item < self.length:
+                line = linecache.getline(self.filename,
+                                         item % len(self) + d)
+                if self.preserve_newline:
+                    return line
+                else:
+                    return line.strip('\r\n')
+
+        elif isinstance(item, slice):
+            low = 0 if item.start is None else item.start
+            low = clip(low, -len(self), len(self) - 1)
+            if low < 0:
+                low += len(self)
+            high = len(self) if item.stop is None else item.stop
+            high = clip(high, -len(self), len(self))
+            if high < 0:
+                high += len(self)
+            ls = []
+            for i in range(low, high):
+                line = linecache.getline(self.filename, i + d)
+                if not self.preserve_newline:
+                    line = line.strip('\r\n')
+                ls.append(line)
+
+            return ls
+
+        raise IndexError('index must be int or slice')
+
+
+class QuestionLoader:
+    def __init__(self, ques_file, tokenizer: QuesNetTokenizer,
+                 pipeline=None, range=None, meta: Optional[list] = None,
+                 content_key=lambda x: x['ques_content'],
+                 meta_key=lambda x: x['know_name'],
+                 answer_key=lambda x: x['ques_answer'],
```

Comment:
`answer_key` and `option_key` seem never use, but it seems import for `__getitem__`